include/net/net_pkt.h

```diff
@@ -1525,6 +1525,16 @@ int net_pkt_copy(struct net_pkt *pkt_dst,
  */
 struct net_pkt *net_pkt_clone(struct net_pkt *pkt, s32_t timeout);
 
+/**
+ * @brief Clone pkt and increase the refcount of its buffer.
+ *
+ * @param pkt Original pkt to be shallow cloned
+ * @param timeout Timeout to wait for free buffer
```

Comment:
"for free packet" the buffer are just refcounted and not allocated

Reply:
True! Updated comment.